From 21de439e0b5f3da0b880c63ea815ae0dab9745bf Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Thu, 11 Aug 2022 16:01:51 +0300
Subject: [PATCH] [SW-81036] boot_desc: use plat specific max num of images

Max number of images is hardcoded locally in the boot desc
implementation file.
Gaudi3 device has larger amount of images in SPI flash due to
additional CPUs such as the PIF ARC.
In order to preserve the memory consumption of the other devices
the max number of images definition is moved to platform specific
header file and larger number of images is given to gaudi3.

Change-Id: I9d614041f747b8f2d0984f232eb46030bc009ff9
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/arch/gaudi/arm/hl.h b/arch/gaudi/arm/hl.h
index 85aaba6..4ba5dc0 100644
--- a/arch/gaudi/arm/hl.h
+++ b/arch/gaudi/arm/hl.h
@@ -6,6 +6,8 @@
 #ifndef ARCH_GAUDI_ARM_HL_H_
 #define ARCH_GAUDI_ARM_HL_H_
 
+#define BOOT_DESC_MAX_IMGS	5
+
 enum hl_rst_src {
 	HL_COLD_RST,
 	HL_MANUAL_RST,
diff --git a/arch/gaudi2/common/hl.h b/arch/gaudi2/common/hl.h
index d608d77..9ae9b6f 100644
--- a/arch/gaudi2/common/hl.h
+++ b/arch/gaudi2/common/hl.h
@@ -9,6 +9,8 @@
 #include <gaudi2/gaudi2_fw_if.h>
 #include "common.h"
 
+#define BOOT_DESC_MAX_IMGS	5
+
 void set_pci_msix_pba_offset_val(void);
 void set_slv_cfg_reg_val(void);
 void halt_n_reset_arc(u8 id);
diff --git a/arch/gaudi3/arc/hl.h b/arch/gaudi3/arc/hl.h
index 4b27668..035d7e3 100644
--- a/arch/gaudi3/arc/hl.h
+++ b/arch/gaudi3/arc/hl.h
@@ -6,6 +6,8 @@
 #ifndef ARCH_GAUDI3_ARC_HL_H_
 #define ARCH_GAUDI3_ARC_HL_H_
 
+#define BOOT_DESC_MAX_IMGS	7
+
 enum gaudi3_rst_src {
 	HL_COLD_RST = 1,
 	HL_MANUAL_RST = 2,
diff --git a/arch/goya/arm/hl.h b/arch/goya/arm/hl.h
index 6c5c535..9bea154 100644
--- a/arch/goya/arm/hl.h
+++ b/arch/goya/arm/hl.h
@@ -6,6 +6,8 @@
 #ifndef ARCH_GOYA_ARM_HL_H_
 #define ARCH_GOYA_ARM_HL_H_
 
+#define BOOT_DESC_MAX_IMGS	5
+
 enum hl_rst_src {
 	HL_COLD_RST,
 	HL_MANUAL_RST,
diff --git a/arch/greco/common/hl.h b/arch/greco/common/hl.h
index ed0458d..e498562 100644
--- a/arch/greco/common/hl.h
+++ b/arch/greco/common/hl.h
@@ -69,6 +69,8 @@
 #define COMMON_HBW_SHRD_NUM	COMMON_SHRD_NUM
 #define COMMON_HBW_SHRD_DIST	COMMON_SHRD_DIST
 
+#define BOOT_DESC_MAX_IMGS	5
+
 struct preboot_sec_rrs_cfg {
 	u32 instances_num;
 	u32 instances_dist;
diff --git a/lib/common/boot_desc.c b/lib/common/boot_desc.c
index 284b5c3..82eb337 100644
--- a/lib/common/boot_desc.c
+++ b/lib/common/boot_desc.c
@@ -12,18 +12,17 @@
 #include <hl_fw_if.h>		/* requires stdint.h */
 #include <hl_boot_desc.h>
 #include <common.h>
+#include <hl.h>
 
 #include "boot_desc.h"
 #include <spi_flash.h>
 #include "uart.h"
 
-#define MAX_IMGS		5
-
 #define BOOT_PTRS_SZ		sizeof(struct hl_fw_boot_ptrs)
 #define IMG_DESC_SZ		sizeof(struct hl_fw_img_desc)
 
 static struct hl_fw_boot_ptrs boot_ptrs;
-static struct hl_fw_img_desc img_desc[MAX_IMGS];
+static struct hl_fw_img_desc img_desc[BOOT_DESC_MAX_IMGS];
 
 static void boot_desc_print_magic(char *name, struct hl_fw_boot_magic *magic)
 {
@@ -59,7 +58,7 @@
 {
 	unsigned int count, i;
 	u32 spi_off = HL_FW_BOOT_PTRS_BASE;
-	u32 desc_offs[MAX_IMGS] = { 0 };
+	u32 desc_offs[BOOT_DESC_MAX_IMGS] = { 0 };
 
 	/* Read the main boot pointers structure */
 	spi_flash_copy(&boot_ptrs, spi_off, BOOT_PTRS_SZ, "BOOT PTRS");
